examples/executor_example/application.hpp

```diff
@@ -0,0 +1,45 @@
+// ---------------------------------------------------------------------------------------------
+// to use a different executor:
+//
+//   1. create a copy of this file in your own application
+//
+//   2. create your own version of appbase/default_executor.hpp
+//           (it needs to define a type `appbase_executor`)
+//      and include this file instead of <appbase/default_executor.hpp>
+//
+//   3. include your own `application.hpp` in your project
+//
+// ---------------------------------------------------------------------------------------------
```

Comment:
Yes, the intent was to have a different executor from the default. I guess I need to finish that. I think it is good to have an example, so that  users don't have to figure things out from the test how to use a different executor. 
What would you prefer, that I finish the exeample with a different executor, or that I remove the example alltogether?

Reply:
These concepts seem generic enough to me to be in appbase.

Reply:
We could provide it as the example maybe?